src/apps/watchfaces/OswAppWatchfaceFitnessAnalog.cpp

```diff
@@ -0,0 +1,323 @@
+/*    
+    printStatus("Temperature", String(hal->environment()->getTemperature() + String("C")).c_str());
+    for(auto& d : *OswTemperatureProvider::getAllTemperatureDevices())
+        printStatus((String("  ") + d->getName()).c_str(), String(d->getTemperature() + String("C")).c_str());
+*/
+    hal->gfx()->setTextSize(2);
+    hal->gfx()->setTextLeftAligned();
+    hal->gfx()->setTextCursor(DISP_W * 0.2f, DISP_H * 0.2f);
+    hal->gfx()->print(hal->environment()->getTemperature(), 1);
+    hal->gfx()->print("C");
+#endif
+}
+
+const char* OswAppWatchfaceFitnessAnalog::getAppId() {
+    return OswAppWatchfaceFitnessAnalog::APP_ID;
+}
+
+const char* OswAppWatchfaceFitnessAnalog::getAppName() {
+    return "AFIT"; //LANG_FITNESS;
+}
+
+void OswAppWatchfaceFitnessAnalog::onStart() {
+    OswAppV2::onStart();
+    OswAppWatchface::addButtonDefaults(this->knownButtonStates);
+
+    // Report that we support short presses on all buttons
+    this->knownButtonStates[Button::BUTTON_SELECT] = ButtonStateNames::SHORT_PRESS ;
+    this->knownButtonStates[Button::BUTTON_UP] = ButtonStateNames::SHORT_PRESS;
+    this->knownButtonStates[Button::BUTTON_DOWN] = ButtonStateNames::SHORT_PRESS;
+
+    // Here is a snippet to also "support" double presses (on BUTTON_SELECT) - note that this WILL DELAY the reporting of any short press events on that button (as it may needs to wait for the second press)
+    this->knownButtonStates[Button::BUTTON_SELECT] = (OswAppV2::ButtonStateNames) (this->knownButtonStates[Button::BUTTON_SELECT] | OswAppV2::ButtonStateNames::DOUBLE_PRESS); // OR to set the bit
+    this->knownButtonStates[Button::BUTTON_UP] = (OswAppV2::ButtonStateNames) (this->knownButtonStates[Button::BUTTON_UP] | OswAppV2::ButtonStateNames::DOUBLE_PRESS); // OR to set the bit
+    this->knownButtonStates[Button::BUTTON_DOWN] = (OswAppV2::ButtonStateNames) (this->knownButtonStates[Button::BUTTON_DOWN] | OswAppV2::ButtonStateNames::DOUBLE_PRESS); // OR to set the bit
+
+    this->lastTime = time(nullptr); // use
+    printf("xxx onStart\n");
+}
+
+void OswAppWatchfaceFitnessAnalog::onLoop() {
+    OswAppV2::onLoop();
+
+    this->needsRedraw = this->needsRedraw or time(nullptr) != this->lastTime; // redraw every second
+}
+
+
+void OswAppWatchfaceFitnessAnalog::onDraw() {
+    OswHal* hal = OswHal::getInstance();
+    hal->setCPUClock(OSW_PLATFORM_DEFAULT_CPUFREQ);
+
+    const int iter = 1;
+    #ifndef OSW_EMULATOR
+        unsigned long old_micros = micros();
+        for (int i=iter; i>0; --i)
+            test();
+        printf("xxxx time for onDraws (average over %d iterations) %f ms.\n", iter, (micros()-old_micros)/(float) iter/1000);
+        Serial.flush();
+    #else
+        unsigned long old_millis = millis();
+        for (int i=iter; i>0; --i)
+            test();
+        printf("xxxx time for onDraws (average over %d iterations) %f ms.\n", iter, (millis()-old_millis)/(float) iter);
+    #endif
+    hal->setCPUClock(20);
```

Comment:
hmmm, slow clock while not drawing - cool idea!

- never reset on stop
- have you measured power usage?
- any effect on e.g. WIFI stuff?
- 20... what is this value? why not slightly less or more? → lowest clock != best power efficiency